airbyte-integrations/connectors/source-google-ad-manager/source_google_ad_manager/streams.py

```diff
@@ -0,0 +1,324 @@
+
+        Args:
+          report_job_id: The ID of the report job to wait for, as a string.
+          export_format: The export format for the report file, as a string.
+          outfile: A writeable, file-like object to write to.
+          include_report_properties: Whether or not to include the report
+            properties (e.g. network, user, date generated...)
+            in the generated report.
+          include_totals_row: Whether or not to include the totals row.
+          use_gzip_compression: Whether or not to use gzip compression.
+        """
+        service = self.report_downloader._GetReportService()
+
+        if include_totals_row is None:  # True unless CSV export if not specified
+            include_totals_row = True if export_format != 'CSV_DUMP' else False
+        opts = {
+            'exportFormat': export_format,
+            'includeReportProperties': include_report_properties,
+            'includeTotalsRow': include_totals_row,
+            'useGzipCompression': use_gzip_compression
+        }
+        report_url = service.getReportDownloadUrlWithOptions(report_job_id, opts)
+        logger.info('Request Summary: Report job ID: %s, %s', report_job_id, opts)
+
+        response = self.report_downloader.url_opener.open(report_url)
+        return response
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        this is the base method to parse the api response
+        """
+        logger.info("start parsing the response stream, should be replaced with logger")
+
+        while True:
+            chunk = response.read(_CHUNK_SIZE)
+            if not chunk:
+                break
+            lines = chunk.decode('utf-8')
+            reader = csv_dict_reader(lines.splitlines())
+            for row in reader:
+                item = self.generate_item(row)
+                yield item.dict()
+    
+    def run_report(self, report_job: str) -> str:
+        """Runs a report, then waits (blocks) for the report to finish generating.
+
+        Args:
+          report_job: The report job to wait for. This may be a dictionary or an
+              instance of the SOAP ReportJob class.
+
+        Returns:
+          The completed report job's ID as a string.
+
+        Raises:
+          An AdManagerReportError if the report job fails to complete.
+        """
+        service = self.report_downloader._GetReportService()
+        report_job_id = service.runReportJob(report_job)['id']
+
+        status = service.getReportJobStatus(report_job_id)
+
+        while status != 'COMPLETED' and status != 'FAILED':
+            logger.debug('Report job status: %s', status)
+            time.sleep(30)
+            status = service.getReportJobStatus(report_job_id)
+
+        if status == 'FAILED':
+            raise AdManagerReportError(report_job_id)
+        else:
+            logger.debug('Report has completed successfully')
+            return report_job_id
```

Comment:
Please create an enum aux class ReportStatus to make this more easy to maintain in the future.

Reply:
👍🏻